awx/sso/backends.py

```diff
@@ -342,7 +342,7 @@ def _update_m2m_from_groups(ldap_user, opts, remove=True):
     if opts is None:
         return None
     elif not opts:
-        pass
+        return None
```

Comment:
Thank you so much for not only pointing this out but also working on fixing it!

In this function `opts` is going to be the `admins` section of the configuration when we are looking at the administrator role. i.e.:
```json
   "admins": [ "cn=group1,ou=groups,dc=example,dc=org", "cn=group2ou=groups,dc=example,dc=org"],
```

Imagine a scenario where someone would want no admins from LDAP and to remove any LDAP users from the admin roles so their setting is:
```json
    "admins": [],
    "remove_admins": true,
```

Now lets say an LDAP user was made an org admin and was logging in, in this scenario they should not be an admin because they are not in `[]` and `remove_admins` is `true`.

The empty list `[]` would trigger this line and `return None` meaning no change to the users permissions. This behavior is incorrect in this scenario.

Instead I think we want to keep this as "pass" to short circuit the remaining if clauses, but then at the bottom of the function instead of just `return False` do something like:
```python
    if remove:
        return False
    return None
```

So if we had no `opts` configuration we would skip the checks that could `return True` and next check if we are supposed to remove users from this role and, if so, `return False` to do that. If we weren't mandated to remove the user and they were not explicitly added than we `return None` so that their permissions for the role do not change.

Please let me know if you agree with that. If so, we can update your PR and keep it moving to get it merged.

Thanks again!

Reply:
Yes, you're right. We need to take the remove status into account, too.

I come up with the following combinations:

| opts     | remove | result (user in opts) | result (user not in opts) |
| ---| --- | --- | ---|
| True     | True   | True                  | ---   |
| True     | False  | True                  | ---   |
| [...]    | True   | True                  | False |
| [...]    | False  | True                  | None  |
| []/False | True   | ---                   | False |
| []/False | False  | ---                   | None  |
| None     | True   | None                  | None  |
| None     | False  | None                  | None  |

I followed your suggestion and updated my code accordingly.